x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/watcher/support/xcontent/XContentSource.java

```diff
@@ -76,8 +77,8 @@ public boolean isMap() {
     /**
      * @return The source as a map
      */
-    public Map<String, Object> getAsMap() {
-        return (Map<String, Object>) data();
+    public Map<?, ?> getAsMap() {
```

Comment:
I understand why you've made this change, but actually I think we want to keep this code the way it was. Can you revert this change please?